### PR TITLE
Added interface for Users and Posts

### PR DESCRIPTION
**Pull request by JustCourt**
Added interfaces IPost and IUser, added functions to IUser. IPost is empty, just added it so I could use it in IUser.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -2,8 +2,9 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/HealthOutLoud.iml" filepath="$PROJECT_DIR$/HealthOutLoud.iml" />
+      <module fileurl="file://C:\Users\court\AndroidStudioProjects\health-out-loud\HealthOutLoud.iml" filepath="C:\Users\court\AndroidStudioProjects\health-out-loud\HealthOutLoud.iml" />
```
Comment (JustCourt) [inline .idea/modules.xml:5]: Android Studio automatically added these lines, and I'm not sure how to make it so they're generic. For now everyone might have to change the filepaths when they update these changes

File: app/src/main/java/com/veephealthoutloud/healthoutloud/Interfaces/IUser.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     * @return true if the post were deleted
+     */
+    boolean DeleteAllUserPosts();
+
+    /**
+     * Report a post.
+     * @param post The post that is reported
+     * @param reason The reason for reporting the post
+     */
+    void ReportPost(IPost post, String reason);
```
Comment (dzhang98) [inline app/src/main/java/com/veephealthoutloud/healthoutloud/Interfaces/IUser.java]: change to boolean? since the delete ones are all boolean.

Comment (JustCourt) [reply app/src/main/java/com/veephealthoutloud/healthoutloud/Interfaces/IUser.java]: good point, changed it to return boolean

**Comment by JustCourt** [comment]
@ArjunM98 @yusufelgharib @dzhang98 If there aren't any issues, i'm going to merge this tomorrow evening so that Yusuf can start writing the IPost methods

**Review by yusufelgharib — APPROVED** [review]

**Review by dzhang98 — APPROVED** [review]

**Pull request merged**